samples/cpp/tutorial_code/videoio/astra/Astra.cpp

```diff
@@ -0,0 +1,158 @@
+#include <opencv2/videoio/videoio.hpp>
+#include <opencv2/highgui.hpp>
+#include <opencv2/imgproc.hpp>
+
+#include <iostream>
+#include <thread>
+#include <condition_variable>
+
+using namespace cv;
+using std::cout;
+using std::cerr;
+using std::endl;
+
+
+int main()
+{
+
+    //! [Open streams]
+    // Open color stream
+    VideoCapture colorStream(CAP_V4L2);
+    // Open depth stream
+    VideoCapture depthStream(CAP_OPENNI2_ASTRA);
```

Comment:
Main problem of this approach is how to sync data from these streams.

Reply:
@alalek, we had a conversation with Orbbec (the producer of Astra cameras), currently this is the only possibility, i.e. the hardware itself does not provide any hardware synchronization and the streaming from both the depth sensor and the color sensor is done absolutely independently.